### DISCUSSION TITLE
Support multiple different operators in the same relation.

### DISCUSSION THREAD

**Discussion by damianh** [openfga/community#12] [Ideas]
(Originally posted to do discord, posting here as requested).

A use case for supporting multiple different operators in the same relation:

Given this model
```
type solution
  relations
    define application-viewer as self
    define owner as self

type application
  relations
    define owner as self
    define blocked as self 
    define viewer as self or application-viewer from owner
```

and the following tuples:

- Solution `foo` has a relation of `owner` with application `bar`. (tuple: "solution:foo", "owner", "application:bar")
- Solution `foo` has a relation of `owner` with application `baz`. (tuple: "solution:foo", "owner", "application:baz")
- An end user `alice` has a relation of `application-view` to solution `foo`. (tuple: "alice", "application-viewer", "solution:foo")

With this, alice as access to all applications in a solution.
- "is alice related to application:bar as viewer?" ✅ 
- "is alice related to application:baz as viewer?" ✅ 

So now I want to explicitly block `alice` from `application:baz`: 
- Add tuple: "alice", "blocked", "application:baz"

Now the model needs to look like:

```
type solution
  relations
    define application-viewer as self
    define owner as self
type application
  relations
    define owner as self
    define blocked as self
    define viewer as self or application-viewer from owner
    define can_view as viewer but not blocked
 ```
 
However the `but not blocked` above is invalid syntax.
 
A work around was provided by @rhamzeh (thanks for quick response there !!) but would be nicer to have support in the DSL.

Thank you!

**Comment by aaguiarz** [comment]
Thanks @damianh, we are considering this change as part of our DSL improvements.

**Comment by aaguiarz** [comment]
This was implemented as part of this PR https://github.com/openfga/language/pull/107
